index.rst

```diff
@@ -1,59 +1,367 @@
+
+.. _Empowerment of DM team members: https://developer.lsst.io/team/empowerment.html#empowerment-of-dm-team-members>
+
+
+Personal and test-package repositories
+======================================
+
+This proposal is primarily concerned with long-lived, shared data repositories of the sort that will exist not just at NCSA, but at the IDF, SLAC, CCIN2P3, and other major LSST data facilities.
+
+Small repositories (typically backed by SQLite) are also expected to be common, especially for small-scale CI and local development.
+These repositories should follow the same naming patterns whenever possible, but will generally not need as many levels of indirection to guard against future changes or collections, and many of the collections defined here as ``CHAINED`` or ``TAGGED`` collections can instead be safely defined directly as ``RUN`` collections instead.
+
+
+Notable omissions and future work
+=================================
+
+"Collections" of data IDs
+-------------------------
+
+Collections that represent fields of particular interest or regularly-reprocessed test datasets are not described here, because those are conceptually more groups of data IDs than groups of datasets (e.g. not just raw exposures, but tracts on which to combine them as well).
+As in Gen2, we will continue to record the definitions of these groups outside the data repository itself, though we may add support for in-repository storage of data IDs to Gen3 in the future.
+It is also worth noting that exposure or visit metadata can sometimes be used to help select some of these data IDs (e.g. ``visit.target_name='SSP-Wide'``), and these selections are automatically combined with the selection of a ``<instrument>/raw/good`` input collection.
+
+
+Naming conventions for dataset types
+------------------------------------
+
+The names for nearly all dataset types in Gen3 have been inherited directly from Gen2, and while these are sorely in need of standardization and cleanup, we have no plans to change to new names until Gen2 has been fully retired.
+Naming conventions for new dataset types would be welcome before then, but are still beyond the scope of this proposal.
+
+In the meantime, users should be aware that dataset types are *global* entities with no implicit namespacing, and hence new dataset types should be created with care.
+The ``pipetask`` tool's ``--register-dataset-types`` option is a non-default option for exactly this reason: in a long-lived repository, re-executions of the same pipeline will eventually outnumber executions of new pipelines (especially new pipelines with new datasets), and hence ``--register-dataset-types`` should rarely be needed.
+Passing it all the time as a matter of habit is an antipattern, because it makes it easy for a typo to result in long-lived, hard-to-clean-up garbage (dataset types can be removed, but only if there are no datasets of that type).
```

Comment:
This seems like something that we should enforce at the code level, but I'm not sure how. Every time you run with `--register-dataset-types` a slack bot sends you a warning? ;-)

Reply:
Objection: we do need this option on all the time in `ap_verify`, because our repositories are (supposed to be) short-lived.

Reply:
One alternative I have been mulling is to make this a separate `pipetask` subcommand, not an option for `pipetask run`.  So you'd do
```
pipetask register-dataset-types -b repo -p mypipeline.yaml
pipetask run -b repo -p mypipeline.yaml ...
```